web/client/components/contextcreator/ConfigureThemes.jsx

```diff
@@ -5,30 +5,73 @@
  * This source code is licensed under the BSD-style license found in the
  * LICENSE file in the root directory of this source tree.
 */
-
-import React from 'react';
+import React, {useRef} from 'react';
 import ReactSelect from 'react-select';
 import {ControlLabel, Glyphicon} from 'react-bootstrap';
+import tinycolor from 'tinycolor2';
+
 import Message from '../I18N/Message';
+import HTML from '../I18N/HTML';
+import Button from '../misc/Button';
+import ColorSelector from '../style/ColorSelector';
+import ToolbarPopover from '../geostory/common/ToolbarPopover';
+import InfoPopover from '../widgets/widget/InfoPopover';
+
+
 import localizedProps from '../misc/enhancers/localizedProps';
-const Select = localizedProps("noResultsText")(ReactSelect);
 import { getMessageById } from '../../utils/LocaleUtils';
 
-const ConfigureThemes = ({
+const Select = localizedProps("noResultsText")(ReactSelect);
+const MAIN_COLOR = "ms-main-color";
+const MAIN_BG_COLOR = "ms-main-bg";
+const PRIMARY = "ms-primary";
+const PRIMARY_CONTRAST = "ms-primary-contrast";
+
+
+function ConfigureThemes({
     themes = [],
     setSelectedTheme = () => {},
     selectedTheme = {},
     disabled = false,
-    context = {}
-}) => {
+    context = {},
+    defaultColors = {
+        [MAIN_COLOR]: "#000000",
+        [MAIN_BG_COLOR]: "#FFFFFF",
+        [PRIMARY]: "#078aa3",
+        [PRIMARY_CONTRAST]: "#FFFFFF"
+    }
+}) {
+    const triggerMain = useRef();
+    const triggerPrimary = useRef();
+    const colors = {
+        ...defaultColors,
+        ...(selectedTheme.variables),
```

Comment:
Take the variables from localConfig mapped to themes props by comparing the id of the selectedTheme